backend/data-store/db-client/store.go

```diff
@@ -64,6 +97,325 @@ func Open(ctx context.Context, log *zap.Logger) (*Store, error) {
+			OrganisationStructure: Denullify(details.OrganisationStructure),
+			DivisionOfWork:        Denullify(details.DivisionOfWork),
+			ProjectSchedule:       Denullify(details.ProjectSchedule),
+			Attatchments:          Denullify(details.Attatchments),
+		},
+	}, nil
+}
+
+func (st *Store) GetSubmissionsByAssessor(ctx context.Context, assessorId int32) (*pb.SubmissionsResponse, error) {
+	returnVal := &pb.SubmissionsResponse{Submissions: []*pb.Submission{}}
+
+	subs, err := queries.New(st.Pool).GetSubmissionsByAssessorId(ctx, assessorId)
+
+	if err == pgx.ErrNoRows {
+		return returnVal, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("getting submissions by assessor id: %w", err)
+	}
+
+	submissions, err := iter.MapErr(subs, func(subm *queries.GetSubmissionsByAssessorIdRow) (*pb.Submission, error) {
+		submission := &pb.Submission{
+			SubmissionId: subm.SubmissionID,
+			Year:         subm.Year,
+			Name:         subm.Name,
+			DurationDays: subm.DurationDays,
+		}
+
+		assessors, errAs := queries.New(st.Pool).GetAssessorsForSubmission(ctx, subm.SubmissionID)
+
+		if errAs != nil && errAs != pgx.ErrNoRows {
+			return nil, fmt.Errorf("getting assessors for submission: %w", errAs)
+		}
+
+		ratings, errRt := queries.New(st.Pool).GetRatingsForSubission(ctx, subm.SubmissionID)
+
+		if errRt != nil && errRt != pgx.ErrNoRows {
+			return nil, fmt.Errorf("getting ratings for submission: %w", errRt)
+		}
+
+		submission.Assessors = MapAssessorsFromSql(assessors)
+		submission.Ratings = MapRatingsFromSql(ratings)
+		return submission, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+	return &pb.SubmissionsResponse{Submissions: submissions}, nil
+}
+
+func (st *Store) GetSubmissionRatings(ctx context.Context, submissionId int32) (*pb.RatingsSubmissionResponse, error) {
+
+	// Run query to get slice of
+	// * AssessorID
+	// * RatingID
+	// * RatingType
+	assessorsWithRating, err := queries.New(st.Pool).GetAssessorsAndRatingsForSubmission(ctx, submissionId)
+
+	if err != nil && err != pgx.ErrNoRows {
+		return nil, fmt.Errorf("getting assessors and ratings for submission: %w", err)
+	}
+
+	var (
+		// Group by rating type ==> map[RatingType]        ->  []struct{ AssessorID, RatingID }
+		// in golang types      ==> map[ProjectRatingType] ->  []*GetAssessorsAndRatingsForSubmissionRow
+		ratingsByType = lo.GroupBy(
+			lo.Map(assessorsWithRating, func(a queries.GetAssessorsAndRatingsForSubmissionRow, _ int) *queries.GetAssessorsAndRatingsForSubmissionRow {
+				return &a
+			}),
+			func(a *queries.GetAssessorsAndRatingsForSubmissionRow) queries.ProjectRatingType {
+				return a.RatingType
+			},
+		)
+
+		// return value
+		returnVal = &pb.RatingsSubmissionResponse{}
+
+		// pool of workers for the rest of logic
+		workers = pool.New().
+			WithContext(ctx).
+			WithMaxGoroutines(4).
+			WithCancelOnError().
+			WithFirstError()
+	)
+
+	// Handle individual ratings
+	workers.Go(func(ctx context.Context) error {
```

Comment:
Starting from here, I think the functions passed to the workers should be extracted. They can be kept within the scope of the parent function, but still, these should be separate function.